src/libasr/codegen/wasm_assembler.h

```diff
@@ -244,4 +86,302 @@ void save_bin(Vec<uint8_t> &code, std::string filename) {
 
 }  // namespace wasm
 
+class WASMAssembler: public WASM_INSTS_VISITOR::WASMInstsAssembler<WASMAssembler> {
+  private:
+    Allocator &m_al;
+
+    Vec<uint8_t> m_type_section;
+    Vec<uint8_t> m_import_section;
+    Vec<uint8_t> m_func_section;
+    Vec<uint8_t> m_memory_section;
+    Vec<uint8_t> m_global_section;
+    Vec<uint8_t> m_export_section;
+    Vec<uint8_t> m_code_section;
+    Vec<uint8_t> m_data_section;
```

Comment:
I think this part should go into WASMFormat class. The WASMAssembler should just create the `m_code_section` part, the rest should be handled by WASMFormat.

Reply:
The `WASMFormat` Class is named as `WASMAssembler` here. The `WASMAssembler` which just creates the `m_code_section` is named as `WASMInstsAssembler`. I will update the names if these are confusing.

Currently, the `WASMAssembler` class inherits from `WASMInstsAssembler` thereby  obtaining/having all the features/powers of creating the wasm format as well as the wasm instructions.

Reply:
Yes, the names are confusing to me. I would keep them completely separate no inheritance. The "assembler" knows how to convert instructions to a binary representation. The "binary format" knows how to create sections, headers, etc. and how to create the final binary. It just copies over the result from the "assembler". If you want, we can discuss this at our meeting more and see what would be a good design.

Reply:
>If you want, we can discuss this at our meeting more and see what would be a good design.

Sure Sir, let us discuss it in the meet.

Reply:
I updated the above comment https://github.com/lcompilers/lpython/pull/1675#discussion_r1161315091.